src/Orleans/Utils/Factory.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Orleans
+{
+    /// <summary>
+    /// Creates an instance of <typeparamref name="TInstance"/>.
+    /// </summary>
+    /// <typeparam name="TInstance"></typeparam>
+    /// <returns></returns>
+    public delegate TInstance Factory<out TInstance>();
```

Comment:
ObjectFactory takes an `IServiceProvider`. Factory<T> is a curried and typed version of it. Check out FactoryUtilities to see that it's just that.
Basically makes the usage of the factory explicit at the consuming level. You just tell it you want to have a factory of an object, you don't care whether it uses SP, ActivatorUtilities, or the new keyword under the hood, you just care to be explicit that you want something to construct new instances of T on demand.